File: nodejs/otc-upload.js
var AWS = require('aws-sdk');

var accessKeyId = process.env.S3_ACCESS_KEY_ID; 
var secretAccessKey = process.env.S3_SECRET_ACCESS_KEY; 
var endpoint = process.env.S3_HOSTNAME; 

AWS.config = new AWS.Config({
  accessKeyId: accessKeyId, secretAccessKey: secretAccessKey, region: 'eu-de'
});

var ep = new AWS.Endpoint(endpoint);
var s3 = new AWS.S3({endpoint: ep});  

// Bucket names must be unique across all S3 users

var myBucket = 'uploadtest';
var myKey = 'uploadtestobject.txt';

s3.createBucket({Bucket: myBucket}, function(err, data) {

if (err) {
   console.log(err);
   } else {
     params = {Bucket: myBucket, Key: myKey, Body: 'Hello!'};
     s3.putObject(params, function(err, data) {
         if (err) {
             console.log(err)
         } else {
             console.log("Successfully uploaded data to uploadtest/uploadtestobject.txt");
         }
      });
   }
});

File: python/otc-upload.py
#!/usr/bin/env python
# -*- coding: utf-8 -*-
# This file is part of OTC S3 Example  released under MIT license.
# Copyright (C) 2016 T-systems Zsolt Nagy
# dir upload - download example: https://github.com/OpenTelekomCloud/python-otcclient/blob/master/otcclient/utils/utils_s3.py

import boto3
from botocore.exceptions import ClientError
import os

ak = os.getenv("S3_ACCESS_KEY_ID", None)
sk = os.getenv("S3_SECRET_ACCESS_KEY", None)       
host = os.getenv("S3_HOSTNAME", None)
region = "eu-de"
		
otcsession = boto3.session.Session()
s3client = otcsession.client('s3', region, 
                           config=boto3.session.Config(signature_version='s3v4'),
                           endpoint_url="https://" + host,
                           aws_access_key_id=ak, 
                           aws_secret_access_key=sk,
                           verify=False
                           )    
myBucket = 'uploadtest'
myKey = 'uploadtestobject.txt'

try:                                   
	s3client.upload_file(myKey, myBucket, "remotefile." + myKey)
except ClientError as e:
    print (str(e))	
except ClientError as e:
	print (str(e))
        
